src/scanelf.d

```diff
@@ -42,152 +42,178 @@ void scanElfObjModule(void delegate(const(char)[] name, int pickAny) pAddSymbol,
     {
         printf("scanElfObjModule(%s)\n", module_name);
     }
-    const buf = base.ptr;
-    const buflen = base.length;
-    int reason = 0;
-    if (buflen < Elf32_Ehdr.sizeof)
+
+    void corrupt(int reason)
     {
-        reason = __LINE__;
-    Lcorrupt:
         error(loc, "corrupt ELF object module %s %d", module_name, reason);
-        return;
     }
-    if (memcmp(buf, elf, 4))
-    {
-        reason = __LINE__;
-        goto Lcorrupt;
-    }
-    if (buf[EI_VERSION] != EV_CURRENT)
+
+    if (base.length < Elf32_Ehdr.sizeof)
+        return corrupt(__LINE__);
+
+    immutable ubyte[4] elf = [0x7F, 'E', 'L', 'F']; // ELF file signature
+    if (memcmp(base.ptr, elf.ptr, elf.length))
+        return corrupt(__LINE__);
+
+    if (base[EI_VERSION] != EV_CURRENT)
     {
-        error(loc, "ELF object module %s has EI_VERSION = %d, should be %d", module_name, buf[EI_VERSION], EV_CURRENT);
+        error(loc, "ELF object module %s has EI_VERSION = %d, should be %d", module_name, base[EI_VERSION], EV_CURRENT);
         return;
     }
-    if (buf[EI_DATA] != ELFDATA2LSB)
+    if (base[EI_DATA] != ELFDATA2LSB)
     {
         error(loc, "ELF object module %s is byte swapped and unsupported", module_name);
         return;
     }
-    if (buf[EI_CLASS] == ELFCLASS32)
+    if (base[EI_CLASS] == ELFCLASS32)
     {
-        Elf32_Ehdr* eh = cast(Elf32_Ehdr*)buf;
+        const eh = cast(const(Elf32_Ehdr)*)base.ptr;
```

Comment:
@andralex This is not the style used in dmd.  Please stop requesting this style in PRs.